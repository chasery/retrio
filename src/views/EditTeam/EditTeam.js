import React, { useEffect, useState } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import TeamApiService from '../../services/teams-api-service';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import FormField from '../../components/FormField/FormField';
import Error from '../../components/Error/Error';
import './EditTeam.css';

function EditTeam(props) {
  const history = useHistory();
  const { teamId } = useParams();
  const [teamName, setTeamName] = useState('');
  const [error, setError] = useState(null);

  useEffect(() => {
    async function initState() {
      try {
        let apiCall = await TeamApiService.getTeam(teamId);
        let res = await apiCall;

        setTeamName(res.name);
      } catch (error) {
        setError(error.error);
      }
    }

    initState();
  }, [teamId]);

  const handleEditTeam = (e) => {
    e.preventDefault();

    TeamApiService.editTeam(teamId, {
      name: teamName,
    })
      .then((res) => history.push(`/teams/${teamId}`))
      .catch((error) => setError(error.error));
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='EditTeam'>
          <div className='EditTeam__wrapper'>
            <Form id='EditTeam' onSubmit={(e) => handleEditTeam(e)}>
              <div className='Form__header'>
                <h2>Edit Team</h2>
              </div>
              <div className='Form__body'>
                <p>
                  Make changes to{' '}
                  {teamName ? <strong>{teamName}</strong> : 'the team'}.
                </p>
                <FormField
                  id='teamName'
                  label='Team Name'
                  type='text'
                  isRequired={true}
                  onChange={(e) => setTeamName(e.target.value)}
                  value={teamName}
                />
                {error ? <Error message={error} /> : null}
                <div className='Form__controls'>
                  <button
                    className='Form__button secondary'
                    type='button'
                    onClick={() => history.push(`/teams/${teamId}`)}
                  >
                    Cancel
                  </button>
                  <button className='Form__button primary' type='submit'>
                    Edit Team
                  </button>
                </div>
              </div>
            </Form>
          </div>
        </section>
      </main>
    </>
  );
}

export default EditTeam;
