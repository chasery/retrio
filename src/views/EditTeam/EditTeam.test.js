import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditTeam from './EditTeam';

describe('EditTeam component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<EditTeam />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
