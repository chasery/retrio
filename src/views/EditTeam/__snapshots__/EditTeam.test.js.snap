// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTeam component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="EditTeam"
    >
      <div
        className="EditTeam__wrapper"
      >
        <Form
          id="EditTeam"
          onSubmit={[Function]}
        >
          <div
            className="Form__header"
          >
            <h2>
              Edit Team
            </h2>
          </div>
          <div
            className="Form__body"
          >
            <p>
              Make changes to 
              <strong>
                Team Name Pending
              </strong>
              .
            </p>
            <FormField
              id="teamName"
              isRequired={true}
              label="Team Name"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="Form__controls"
            >
              <button
                className="Form__button secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="Form__button primary"
                type="submit"
              >
                Edit Team
              </button>
            </div>
          </div>
        </Form>
      </div>
    </section>
  </main>
</Fragment>
`;
